Plug-Ins/NovaTools/System Editor/StellarView.swift

```diff
@@ -0,0 +1,211 @@
+import AppKit
+import RFSupport
+
+class StellarView: NSView {
+    let resource: Resource
+    let manager: RFEditorManager
+    let isEnabled: Bool
+    private(set) var position = NSPoint.zero
+    private(set) var image: NSImage? = nil
+    private var attributes: [NSAttributedString.Key: Any] {
+        [.foregroundColor: NSColor.white, .font: NSFont.systemFont(ofSize: 11)]
+    }
+    private var systemView: SystemMapView? {
+        superview as? SystemMapView
+    }
+    var point = NSPoint.zero {
+        didSet {
+            frame.origin.x = point.x - frame.size.width / 2
+            frame.origin.y = point.y - frame.size.height / 2
+        }
+    }
+    var isHighlighted = false {
+        didSet {
+            if oldValue != isHighlighted {
+                needsDisplay = true
+            }
+        }
+    }
+    var highlightCount = 1 {
+        didSet {
+            if oldValue != highlightCount {
+                needsDisplay = true
+            }
+        }
+    }
+
+    init?(_ stellar: Resource, manager: RFEditorManager, isEnabled: Bool) {
+        self.resource = stellar
+        self.manager = manager
+        self.isEnabled = isEnabled
+        super.init(frame: .zero)
+        do {
+            try self.read()
+        } catch {
+            return nil
+        }
+    }
+
+    func read() throws {
+        let reader = BinaryDataReader(resource.data)
+        position.x = Double(try reader.read() as Int16)
+        position.y = Double(try reader.read() as Int16)
+
+        let spinId = Int(try reader.read() as Int16) + 1000
+        if let spinResource = manager.findResource(type: ResourceType("spïn"), id: spinId, currentDocumentOnly: false) {
+            let spinReader = BinaryDataReader(spinResource.data)
+            let spriteId = (Int)(try spinReader.read() as Int16);
+
+            // Use the preview provider to obtain an NSImage from the resources the spïn might point to -- try rlëD then fall back to PICT
+            if let spriteResource = manager.findResource(type: ResourceType("rlëD"), id: spriteId, currentDocumentOnly: false) ?? manager.findResource(type: ResourceType("PICT"), id: spriteId, currentDocumentOnly: false) {
+                spriteResource.preview { img in
+                    self.image = img
```

Comment:
I think this might need a `needsDisplay`? It doesn't seem to show the image for me on first load.

Reply:
I realised the needsDisplay was just misplaced. I pushed a fix and also separated the load graphic into a function so errors can be ignored (e.g. if the spin was empty for some reason).